csharp-basic-topics/GuidClassInCSharp/GuidClassInCSharpTests/GuidClassInCSharpUnitTests.cs

```diff
@@ -0,0 +1,54 @@
+using static GuidClassInCSharp.GuidClassInCSharpMethods;
+
+namespace GuidClassInCSharpTests
+{
+    public class Tests
+    {
+        [Test]
+        public void GivenCheckGuidEqualityMethod_WhenEqualGuids_ThenTrue()
+        {
+            var guidA = new Guid("B5724FB4-2FA0-4E68-B108-93C24A1EA6DF");
+            var guidB = new Guid("B5724FB4-2FA0-4E68-B108-93C24A1EA6DF");
+            var checkEquality = CheckGuidEquality(guidA, guidB);
+
+            Assert.That(checkEquality, Is.EqualTo(true));
+        }
+
+        [Test]
+        public void GivenCheckGuidEqualityMethod_WhenNotEqualGuids_ThenFalse()
+        {
+            var guidA = Guid.NewGuid();
+            var guidB = Guid.NewGuid();
+            var checkEquality = CheckGuidEquality(guidA, guidB);
+
+            Assert.That(checkEquality, Is.EqualTo(false));
+        }
+
+        [Test]
+        public void GivenCreateEmptyGuidMethod_WhenCalled_ThenGuidEmpty()
+        {
+            var emptyGuid = new Guid("00000000-0000-0000-0000-000000000000");
+            var createdGuid = CreateEmptyGuid();
+
+            Assert.That(emptyGuid, Is.EqualTo(createdGuid));
+        }
+
+        [Test]
+        public void GivenCreateRandomGuidMethod_WhenCalled_ThenValidGuid()
```

Comment:
For completeness this should maybe have an additional Assert that the created random GUID is not the empty GUID